File: programmers/hash_2.java
//전화번호 목록
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        HashMap<String,Integer> map = new HashMap<>();
        
        //해시맵의 키에 값을 저장하고 value에 인덱스를 부여
        for(int i = 0;i<phone_book.length;i++){
            map.put(phone_book[i],i);
        }
        
        for(int k = 0;k<phone_book.length;k++){
            for(int j = 1;j<phone_book[k].length();j++){
                if(map.containsKey(phone_book[k].substring(0,j))){
                    answer = false;
                    return answer;
                }
            }
        }
        
        return answer;
    }
}

File: programmers/hash_2.py
#전화번호 목록
def solution(phone_book):
    answer = True
    dic = {}
    #딕셔너리 키에 값(전화번호)를 저장하고 value에 인덱스를 부여
    for i in range(len(phone_book)):
        dic[phone_book[i]] = i
    for i in range(len(phone_book)):
        for j in range(1,len(phone_book[i])):
            if(phone_book[i][0:j] in dic):
                answer = False
    return answer
